src/proxy/processors/push-action/parsePush.ts

```diff
@@ -93,30 +103,79 @@ async function exec(req: any, action: Action): Promise<Action> {
     const [meta, contentBuff] = getPackMeta(buf);
     const contents = getContents(contentBuff as any, meta.entries as number);
 
-    action.commitData = getCommitData(contents as any);
-    if (action.commitData.length === 0) {
-      step.log('No commit data found when parsing push.');
-    } else {
+    const ParsedObjects = {
+      commits: [] as CommitData[],
+      tags: [] as TagData[],
+    };
+
+    for (const obj of contents) {
+      if (obj.type === GIT_OBJECT_TYPE_COMMIT)
+        ParsedObjects.commits.push(...getCommitData([obj]));
+      else if (obj.type === GIT_OBJECT_TYPE_TAG) ParsedObjects.tags.push(parseTag(obj));
+    }
+
+    action.commitData = ParsedObjects.commits;
+    action.tagData = ParsedObjects.tags;
+
+    if (action.commitData.length) {
       if (action.commitFrom === EMPTY_COMMIT_HASH) {
         action.commitFrom = action.commitData[action.commitData.length - 1].parent;
       }
-      const user = action.commitData[action.commitData.length - 1].committer;
-      action.user = user;
+      action.user = action.commitData.at(-1)!.committer;
+    } else if (action.tagData?.length) {
+      action.user = action.tagData.at(-1)!.tagger;
+    } else {
+      throw new Error('No commit or tag data parsed from packfile');
     }
-
     step.content = {
       meta: meta,
     };
   } catch (e: any) {
     step.setError(
-      `Unable to parse push. Please contact an administrator for support: ${e.toString('utf-8')}`,
+      `Unable to parse push. Please contact an administrator for support: ${e.message || e.toString()}`,
     );
   } finally {
     action.addStep(step);
   }
   return action;
 }
 
+function parseTag(x: CommitContent): TagData {
```

Comment:
I noticed that this function is not extracting the timestamp, which is actually present in the `tagger` field within the `content`:

<img width="945" height="428" alt="Image" src="https://github.com/user-attachments/assets/6f1b7016-39dd-4946-9c47-c011f9ef3dd4" />

We should extract it and pass it to the frontend, which currently displays N/A:

<img width="3147" height="275" alt="Image" src="https://github.com/user-attachments/assets/d1086338-d7c0-4941-83a6-fc2da2cadf69" />